src/main/java/chess/dao/StateDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package chess.dao;
+
+import chess.dao.converter.StateToStringConverter;
+import chess.dao.converter.StringToStateConverter;
+import chess.model.board.Board;
+import chess.model.state.State;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class StateDaoImpl implements StateDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public StateDaoImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public void insert(final State state) {
+        final String sql = "insert into state (name) values (?)";
+        final String stateName = StateToStringConverter.convert(state);
+        jdbcTemplate.update(sql, stateName);
+    }
+
+    public State find(final Board board) {
+        final String sql = "select name from state";
+        String stateName = jdbcTemplate.queryForObject(sql, String.class);
+        return StringToStateConverter.convert(stateName, board);
+    }
+
+    public int delete() {
```

Comment:
```suggestion
    public int deleteAll() {
```

where를 생략하면 모든 데이터가 삭제되니 메서드명에 표시해주세요.

Reply:
현재는 `id`를 적용하면서 delete로 사용하는것이 적절하도록 하였어요!